### PR TITLE
unit testing

### PR DESCRIPTION
**Pull request by Matias-Villalobo**
As you can see, the corresponding tests have been carried out

image

### CODE REVIEW

File: app/src/test/java/com/example/mykotlincalculator21/mvp/presenter/CalculatorPresenterTest.kt [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+import com.example.mykotlincalculator21.utils.StringUtilsTest.NUMBER_TWO
+import com.example.mykotlincalculator21.utils.StringUtilsTest.NUMBER_ZERO
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import junit.framework.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+
+class CalculatorPresenterTest {
+    private lateinit var presenter: CalculatorPresenter
```
Comment (SartiMau) [inline app/src/test/java/com/example/mykotlincalculator21/mvp/presenter/CalculatorPresenterTest.kt]: remember to use the contract type pls

File: app/src/test/java/com/example/mykotlincalculator21/mvp/presenter/CalculatorPresenterTest.kt [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+    private var model = CalculatorModel()
```
Comment (SartiMau) [inline app/src/test/java/com/example/mykotlincalculator21/mvp/presenter/CalculatorPresenterTest.kt]: add the contract type here pls

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/test/java/com/example/mykotlincalculator21/mvp/presenter/CalculatorPresenterTest.kt (hunk 2 above)
Comment (Matias-Villalobo) [reply app/src/test/java/com/example/mykotlincalculator21/mvp/presenter/CalculatorPresenterTest.kt]: Done thx

File: app/src/test/java/com/example/mykotlincalculator21/mvp/presenter/CalculatorPresenterTest.kt (hunk 1 above)
Comment (Matias-Villalobo) [reply app/src/test/java/com/example/mykotlincalculator21/mvp/presenter/CalculatorPresenterTest.kt]: Done thx

**Review by SartiMau — APPROVED** [review]

File: app/src/test/java/com/example/mykotlincalculator21/mvp/presenter/CalculatorPresenterTest.kt [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+
+    @Before
+    fun setup() {
+        model = CalculatorModel()
+        presenter = CalculatorPresenter(model, view)
+    }
+
+    @Test
+    fun eraseTest() {
+        presenter.erase()
```
Comment (lucianoeli) [inline app/src/test/java/com/example/mykotlincalculator21/mvp/presenter/CalculatorPresenterTest.kt:38]: you should set the model with data, to test if it really delete the model. cause by default i assume is empty

File: app/src/test/java/com/example/mykotlincalculator21/mvp/presenter/CalculatorPresenterTest.kt [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+    fun testOperationSymbolPressedPlus() {
+        model.saveNumber(NUMBER_THREE)
+        presenter.operationSymbolPressed(OPERATOR_SUM)
+        verify(view).drawNumber(NUMBER_THREE + OPERATOR_SUM)
+        assertEquals(NUMBER_THREE + OPERATOR_SUM, model.getValue())
+    }
+
+    @Test
+    fun testOperationSymbolPressedNegative() {
+        model.saveNumber(NUMBER_THREE)//ojo aca
```
Comment (lucianoeli) [inline app/src/test/java/com/example/mykotlincalculator21/mvp/presenter/CalculatorPresenterTest.kt]: please delete comment

File: app/src/test/java/com/example/mykotlincalculator21/mvp/presenter/CalculatorPresenterTest.kt [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+        presenter.numberPressed(NUMBER_THREE)
+        verify(view).drawNumber(model.getValue())
+        assertEquals(NUMBER_THREE + OPERATOR_SUM + NUMBER_THREE, model.getValue())
+    }
+
+    @Test
+    fun testSecondOperandNegativePressed() {
+        model.saveNumber(NUMBER_THREE)
+        model.saveOperationSymbol(OPERATOR_SUM)
+        presenter.numberPressed(OPERATOR_MINUS + NUMBER_THREE)
```
Comment (lucianoeli) [inline app/src/test/java/com/example/mykotlincalculator21/mvp/presenter/CalculatorPresenterTest.kt]: is it ok that the presenter is called with (-3)?. I think it should be called twice, first with - and then with 3

File: app/src/test/java/com/example/mykotlincalculator21/mvp/presenter/CalculatorPresenterTest.kt (hunk 4 above)
Comment (Matias-Villalobo) [reply app/src/test/java/com/example/mykotlincalculator21/mvp/presenter/CalculatorPresenterTest.kt]: Done thx

File: app/src/test/java/com/example/mykotlincalculator21/mvp/presenter/CalculatorPresenterTest.kt (hunk 5 above)
Comment (Matias-Villalobo) [reply app/src/test/java/com/example/mykotlincalculator21/mvp/presenter/CalculatorPresenterTest.kt]: Done thx

File: app/src/test/java/com/example/mykotlincalculator21/mvp/presenter/CalculatorPresenterTest.kt (hunk 3 above)
Comment (Matias-Villalobo) [reply app/src/test/java/com/example/mykotlincalculator21/mvp/presenter/CalculatorPresenterTest.kt:38]: Done thx

**Review by BQuinteros — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Pull request merged**